dafny-lang/dafny
# Standard lib support in `dafny server`

### Description
- Add support for the Dafny standard library to `dafny server`
- Add support for `library = [..]` in `dfyconfig.toml` to `dafny server`
- Refactor error handling for errors in dfyconfig.toml, so it is consistent between `dafny server` and other commands.
- Ensure `dafny server` only publishes notifications for Uris owned by a particular project. A project owns a Uri if it is the closest project to that Uri and it includes it.
- Moves code in `new Dafny` to `DafnyFile.CreateAndValidate`, so it's more clear that it contains logic.
- Refactor the logic in `DafnyFile` so it reports errors using an `ErrorReporter` instead of stdio, which makes it reusable for the language server.
- Stop using the exception `IllegalDafnyFile` to handle common scenarios.
- Replace obsolete usage of `Lazy<IdeState>` in `ProjectManager` with `IdeState`

### How has this been tested?
- Add a test for using the standard library with `dafny server`
- Add a test for using an unfound library file with `dafny server`
- Add a test for using a library file with `dafny server`

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>
